pkg/kube_events_manager/monitor.go

```diff
@@ -160,10 +161,13 @@ func (m *monitor) CreateInformers() error {
 				for _, informer := range m.VaryingInformers[nsName] {
 					informer.WithContext(ctx)
 					informer.Start()
+					if m.eventsEnabled {
+						informer.EnableKubeEventCb()
+					}
```

Comment:
For me it is not too much clear, why `CreateInformers` can be called multiple times. 

Probably there could be single config on the start, and it should be called only once per application lifecycle. Seems that is not the case.

Then, since this method is public, I thought it could be called by clients. Bu I think it is only called on the addition of configs when hooks are initialized. And the call is hidden under Monitor interface. That became cleaner to me.

What if `CreateInformers` was private? Can we make it private, assuming nobody commited to use this interface?

Reply:
`CreateInformers` should be called once, it is more an `Init` method that configure monitor and may return error. If this initialization was successful, monitor can be `Start`ed and `Stop`ed multiple times.

`informer.EnableKubeEventCb()` is used inside an Add/Modify callback for the namespace informer. So this block can be called multiply times on changes in namespace labels.

Reply:
`monitor.EnableKubeEventCb()` is called after successful hook execution with Synchronization context.

`monitor.CreateInformers` creates an informer for Namespaces if NamespaceSelector.LabelSelector is defined in MonitorConfig. This informer has two callbacks for Add/Modify and Delete events from namespaces. So, `monitor.eventsEnabled` will be checked every time when new namespace added or deleted, not once during `monitor.CreateInformers` call.